src/ui/src/core/index.ts

```diff
@@ -709,6 +725,30 @@ export function generateCore() {
 		setTimeout(() => checkIfStateEnquiryRequired(0), INITIAL_FOLLOWUP_MS);
 	}
 
+	function waitSyncHealthToBe(value: SyncHealth) {
+		if (syncHealth.value === value) return;
+
+		// eslint-disable-next-line no-async-promise-executor
+		return new Promise<void>(async (res) => {
+			while (syncHealth.value !== value) {
+				await new Promise((r) => setTimeout(r, 500));
+			}
+			return res();
+		});
+	}
+
+	async function sendFrontendMessageQueue(
+		type: string,
+		payload: object | (() => Promise<object>),
+		callback?: Function,
+		track = false,
+	) {
+		return queue.add(async () => {
+			await waitSyncHealthToBe("connected");
+			return sendFrontendMessage(type, payload, callback, track);
+		});
```

Comment:
It seems you don't need to return anything.
```suggestion
		queue.add(async () => {
			await waitSyncHealthToBe("connected");
			sendFrontendMessage(type, payload, callback, track);
		});
```

Reply:
It is not equivalent.

`queue.add` is adding the async func to the queue and running it. I omit to `return sendFrontendMessage()`, it won't wait and return before the message is sent.